components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/LoginContextManagementUtil.java

```diff
@@ -316,4 +319,14 @@ private static boolean isStepHasMultiOption(AuthenticationContext context) {
         }
         return stepHasMultiOption;
     }
+
+    private static boolean isValidRequest(HttpServletRequest request, AuthenticationContext context) {
+
+        boolean isValidRequest = context != null;
+        if (isNonceCookieEnabled() && isValidRequest) {
+            Cookie nonceCookie = FrameworkUtils.getCookie(request, getNonceCookieName(context));
```

Comment:
let's add a debug log here

Reply:
Fixed with 08be66bdf52f81ddf594faeced9e21e1917cc4d7. Added to the regular flow rather to the utility method, as we are already checking the validity status there.